### PR TITLE
Improve compiler compatibility

### PR DESCRIPTION
**Pull request by krlmlr**
- Undeclared winapi
- Uninitialized move
- clang19 `-Wmissing-template-arg-list-after-template-kw`

For #12385.

### CODE REVIEW

File: src/include/duckdb/function/scalar/regexp.hpp [hunk 1]
```diff
@@ -106,9 +106,12 @@ struct RegexStringPieceArgs {
 	}
 
 	RegexStringPieceArgs &operator=(RegexStringPieceArgs &&other) noexcept {
-		std::swap(this->size, other.size);
-		std::swap(this->capacity, other.capacity);
-		std::swap(this->group_buffer, other.group_buffer);
+		this->size = other.size;
```
Comment (hannes) [inline src/include/duckdb/function/scalar/regexp.hpp:109]: What's wrong with `std::swap` here?

**Comment by hannes** [comment]
LGTM otherwise

File: src/include/duckdb/function/scalar/regexp.hpp (hunk 1 above)
Comment (krlmlr) [reply src/include/duckdb/function/scalar/regexp.hpp:109]: Good question, let me check.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**